mast/tools/sis_authentication.py

```diff
@@ -0,0 +1,118 @@
+
+
+def authenticate_via_sis(name, surname, login=None, ukco=None, is_employee=False):
+    """
+    Simple authentication via sis.
+    Employee must submit: name, surname
+    Student must submit: name, surname, (login or ukco)
+    :param name: name of the person - obligatory:
+    :param surname: surname of the person - obligatory:
+    :param login: login for students
+    :param ukco: UKCO for students
+    :param is_employee: bool if the person is employee
+    :return: True if such person exists, False otherwise
+    """
+    if is_employee:
+        if name == None or surname == None:
+            # expecting both name and surname
+            print('Expected both name and surname\nName={}\nSurname={}'.format(name, surname))
+            return False
+        return authenticate_employee(name=name, surname=surname)
+    else:
+        if name == None or surname == None:
+            # expecting both name and surname
+            print('Expected both name and surname\nName = {}\nSurname = {}'.format(name, surname))
+            return False
+        if login == None and ukco == None:
+            # expecting at least one of the following login, ukco
+            print('Expected at least one of login, ukco\nLogin = {}\nUKCO = {}'.format(login, ukco))
+            return False
+        return authenticate_student(name=name, surname=surname, login=login, ukco=ukco)
+
+
+def authenticate_student(name, surname, login=None, ukco=None):
+    url = build_url(is_employee=False, name=name, surname=surname, login=login, ukco=ukco)
+    page = requests.get(url)
+    nubmer_of_results = get_number_of_students(page=page, name=name, surname=surname, login=login, ukco=ukco)
+
+    if int(nubmer_of_results) >= 1:
+        return True
+    return False
+
+
+def authenticate_employee(name, surname):
+    url = build_url(is_employee=True, name=name, surname=surname)
+    page = requests.get(url)
+    nubmer_of_results = get_number_of_employees(page=page, name=name, surname=surname)
+
+    if int(nubmer_of_results) >= 1:
+        return True
+    return False
+
+
+def get_number_of_employees(page, name, surname):
+    soup = BeautifulSoup(page.content, 'html.parser')
+    content = soup.select('#page_div > b:nth-child(3)')
+    for text in content:
+        for part in text:
+            return get_number(part)
+    return 0
+
+
+def get_number_of_students(page, name, surname, login=None, ukco=None):
+    soup = BeautifulSoup(page.content, 'html.parser')
+    content = soup.select('#content > table >  tr > td.info_text > ul > li')
+    for text in content:
+        for part in text:
+            if ("Number of students who do not have the right to seek:" in part) or \
+                    ("Počet studentů, které nemáte právo vyhledat:" in part):
+                return get_number(part)
+    return 0
+
+
+def get_number(text):
+    return int(re.search(r'\d+', text).group())
+
+
+def build_url(is_employee, name=None, surname=None, login=None, ukco=None):
```

Comment:
Consider using prefix '__' in order to make a function private and hence avoid injection to the attributes.

Reply:
Isnt __ prefix only for objects? this is plain function